components/python/tox/Makefile

```diff
@@ -1,85 +1,37 @@
+# This file was automatically generated using the following command:
+#   $WS_TOOLS/python-integrate-project tox
 #
 
-BUILD_STYLE= setup.py
-include ../../../make-rules/shared-macros.mk
+BUILD_STYLE = setup.py
 
-COMPONENT_NAME=		tox
-COMPONENT_VERSION=	3.25.1
-COMPONENT_SUMMARY=	Command line driven CI frontend and development task automation tool
-COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
-COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
-COMPONENT_ARCHIVE_HASH= sha256:c138327815f53bc6da4fe56baec5f25f00622ae69ef3fe4e1e385720e22486f9
-COMPONENT_ARCHIVE_URL=	$(call pypi_url)
-COMPONENT_PROJECT_URL=	https://tox.wiki/
-COMPONENT_FMRI=		library/python/tox
-COMPONENT_CLASSIFICATION=	Development/Python
-COMPONENT_LICENSE=	MIT
+include ../../../make-rules/shared-macros.mk
 
-# tox is a tool so we provide only a single package
-PYTHON_VERSIONS=	3.9
-PYTHON_VERSION=		3.9
+COMPONENT_NAME =		tox
+COMPONENT_VERSION =		3.26.0
+COMPONENT_SUMMARY =		tox - tox is a generic virtualenv management and test command line tool
+COMPONENT_PROJECT_URL =		https://tox.readthedocs.io
+COMPONENT_ARCHIVE_URL =		\
+	https://files.pythonhosted.org/packages/59/14/1931cd6cb9804f2326cccdd750af32a5095132b7d73cc33577644ab59888/tox-3.26.0.tar.gz
+COMPONENT_ARCHIVE_HASH =	\
+	sha256:44f3c347c68c2c68799d7d44f1808f9d396fc8a1a500cbc624253375c7ae107e
+COMPONENT_LICENSE =		MIT
+COMPONENT_LICENSE_FILE =	LICENSE
 
 include $(WS_MAKE_RULES)/common.mk
 
-COMPONENT_TEST_DIR =	$(SOURCE_DIR)/tests
-COMPONENT_TEST_CMD =	$(PYTHON) -m pytest
-COMPONENT_TEST_ARGS =
-COMPONENT_TEST_MASTER = $(COMPONENT_TEST_RESULTS_DIR)/results-all.master
-COMPONENT_TEST_TRANSFORMS += \
-	'-e "s/in [0-9]*\.[0-9]*s //"' \
-	'-e "s/[0-9]*:[0-9]*:[0-9]*//"' \
-	'-n' \
-	'-e "/===/p"' \
-	'-e "/collected/p"' \
-	'-e "/^ERROR/p"' \
-	'-e "/^FAILED/p"' \
-	'-e "/^SKIPPED/p"' \
-	'-e "/^test_/p"'
-
-ASLR_MODE = $(ASLR_NOT_APPLICABLE)
-
-# Test note: library/python/pytest-randomly-39 let the tests fail because of internal errors.
-
-# Manually added test dependencies according to setup.cfg
-REQUIRED_PACKAGES += library/python/filelock-39
-REQUIRED_PACKAGES += library/python/flaky-39
-REQUIRED_PACKAGES += library/python/freezegun-39
-REQUIRED_PACKAGES += library/python/hypothesis-39
-REQUIRED_PACKAGES += library/python/pluggy-39
-REQUIRED_PACKAGES += library/python/psutil-39
-REQUIRED_PACKAGES += library/python/py-39
-REQUIRED_PACKAGES += library/python/pytest-39
-REQUIRED_PACKAGES += library/python/pytest-benchmark-39
-REQUIRED_PACKAGES += library/python/pytest-cov-39
-REQUIRED_PACKAGES += library/python/pytest-mock-39
-# Our pytest-randomly doesn't seem to work
-#REQUIRED_PACKAGES += library/python/pytest-randomly-39
-REQUIRED_PACKAGES += library/python/pytest-reporter-39
-REQUIRED_PACKAGES += library/python/virtualenv-39
-
 # Auto-generated dependencies
-REQUIRED_PACKAGES += library/python/setuptools-39
-REQUIRED_PACKAGES += runtime/python-39
+PYTHON_REQUIRED_PACKAGES += library/python/importlib_metadata
```

Comment:
What are your plans regarding metadata use from setup.py and setup.cfg? With your changes we will loose these manually added dependencies.
With your changes we now hav 0 tests are being run, formerly we had
 18 failed, 613 passed, 5 skipped, 1 warning

Reply:
I plan to implement fully automatic dependency discovery so the final result is something similar as we do have with perl modules now.  But first I'd like to finish support for tox based testing.

Yes, previously tox used `pytest` (as suggested by `tox.ini`).  This PR switches to `setup.py test` which is currently the only supported automatic testing we do have.  As you properly noticed, tox does not support `setup.py test`, so we ended with `Ran 0 tests` result.  This is interim.  Once I implement (fully automatic) tox based testing this (and many other python projects too) will switch to tox testing and we will get real test results back.